clippy_lints/src/utils/conf.rs

```diff
@@ -402,6 +402,10 @@ define_Conf! {
     /// A list of paths to types that should be treated like `Arc`, i.e. ignored but
     /// for the generic parameters for determining interior mutability
     (ignore_interior_mutability: Vec<String> = Vec::from(["bytes::Bytes".into()])),
+    /// Lint: INDEXING_SLICING
+    ///
+    /// Whether to suppress lint in const function
+    (suppress_lint_in_const: bool = true),
```

Comment:
I would include the lint category in the lint name, as it could be possible, that users want restriction lints to trigger in every context while style lints, only if it can be rewritten.

Then we could also think about defaulting this to false, but having it on true is the safe choice :+1:

Reply:
sorry i dont quite get it, do you mean the name should be something like `suppress_{category}_lint_in_const`?